### DISCUSSION TITLE
2022 Plan

### DISCUSSION THREAD

**Discussion by koh1project** [koh1project/apprenticeship_to_master#17] [Ideas]
# Learning plan of 2022

### 1. MERN Stack development
Typical MERN Stack development with several ideas and challenges.
  
There are 2 things what I want to challenge:
- To research for my question: "Is Apollo Client a good option global state management tool?"  
Apollo Client is used  as GraphQL client tool, and can be used as state management tool, however, I have watched and heard some projects use other global state management tool, such as Redux or Recoil, with Apollo Client. I wondered why they take the architecture, so I would like to use Apollo Client as both GraphQL tool and state management tool.
  
- Write auto-testing, and try TDD process partially
Try to write testing code that I have never used and take TDD process, at least 1 front-end page, and 1 back-end function.

### 2. Job hunting preparation phase
Focus on the Job hunting process, Resume brush up, interview prep, English, etc...
  
## How to manage
- I set the candidate tech stack in the Issues of this repository, and pick them, some of them will not be used and labelled such as 'wontfix.'
- Set sprint for each week
- Project Cycle in the Issue property stands for each month

**Comment by koh1project** [comment]
It seems 1 month for MERN stack development is tight schedule and may need to cut CI/CD or deployment on a cloud environment.

**Comment by kei95** [comment]
Hey, Good stuff! it must've been quite a work to set everything up here. One thing I'd recommend is to avoid setting "deadlines" in your study plans because it has a high chance to affect the quality of the learning. It's easy to miss-estimate time you would need for each material so it gets you under unnecessary pressure and might even end up sacrificing the quality of study just to meet it.
Overall though, I think it's a good idea to build a simple website to try out new technologies. There are some items that could be coupled together so I'll comment on these tickets. Good job!🎉

**Reply by koh1project** [reply]
Thank you, Kei for checking all things. Sure, I can totally agree with your suggestion and removed the schedule restriction. I will check the comments on the tickets and change them!
